### PR TITLE
Providing additional printer

### PR DESCRIPTION
**Pull request by ReigenDing**
Add new addtional printer columns for add chaos crds.

### What problem does this PR solve?
providing additional printer columns `action` & `duration` for each type of Chaos. #2440 

> Problem Summary:

### What's changed and how it works?

> What's Changed:

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Chaos Dashboard`
- [ ] Need to **cheery-pick to release branches**

### Checklist

Tests

- [x] Unit test
- [x] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```
providing additional printer columns `action` & `duration` for each type of Chaos

### CODE REVIEW

**Comment by STRRL** [comment]
Hi @ReigenDing , I am afraid that we could not modify the `crd.yaml` manually. 

We generate `crd.yaml` with `controller-gen`(https://github.com/kubernetes-sigs/controller-tools) so you need to add some comments on the Go Struct in `api/v1alph1`

**Comment by ReigenDing** [comment]
Hi @STRRL,  I try to fix it according you guide.Please let me known if it is ok.

**Comment by STRRL** [comment]
> Hi @STRRL, I try to fix it according to you guide.Please let me known if it is ok.

Please update the yaml files by executing `make check` on your local machine, it will re-generate the CRD yaml definitions. :D

File: api/v1alpha1/httpchaos_types.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 )
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/httpchaos_types.go:24]: `HTTPChaos` does not contains a field `.spec.action`

File: api/v1alpha1/kernelchaos_types.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import (
 )
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/kernelchaos_types.go:24]: `KernelChaos` does not contains a field .spec.action

File: api/v1alpha1/stresschaos_types.go [hunk 3]
```diff
@@ -25,6 +25,8 @@ import (
 // Stress chaos is a chaos to generate plenty of stresses over a collection of pods.
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/stresschaos_types.go:29]: `StressChaos` does not contains a field .spec.action

File: api/v1alpha1/timechaos_types.go [hunk 4]
```diff
@@ -20,6 +20,8 @@ import (
 )
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
```
Comment (STRRL) [inline api/v1alpha1/timechaos_types.go:23]: `TimeChaos` does not contains a field .spec.action

File: api/v1alpha1/podhttpchaos_types.go [hunk 5]
```diff
@@ -182,6 +182,8 @@ const (
 )
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/podhttpchaos_types.go:186]: We do not need add this feature for `PodXXXChaos`

File: api/v1alpha1/podiochaos_types.go [hunk 6]
```diff
@@ -243,6 +243,8 @@ type PodIOChaosStatus struct {
 }
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/podiochaos_types.go:247]: ditto, We do not need add this feature for PodXXXChaos

File: api/v1alpha1/podnetworkchaos_types.go [hunk 7]
```diff
@@ -23,6 +23,8 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="action",type=string,JSONPath=`.spec.action`
+// +kubebuilder:printcolumn:name="duration",type=string,JSONPath=`.spec.duration`
```
Comment (STRRL) [inline api/v1alpha1/podnetworkchaos_types.go:27]: ditto, We do not need add this feature for PodXXXChaos

**Review by STRRL — CHANGES_REQUESTED** [review]
PTAL @ReigenDing

File: api/v1alpha1/httpchaos_types.go (hunk 1 above)
Comment (ReigenDing) [reply api/v1alpha1/httpchaos_types.go:24]: So I need to remove this field or change the XXXXChaosSpec in that situation?

File: api/v1alpha1/podhttpchaos_types.go (hunk 5 above)
Comment (ReigenDing) [reply api/v1alpha1/podhttpchaos_types.go:186]: OK

File: api/v1alpha1/httpchaos_types.go (hunk 1 above)
Comment (STRRL) [reply api/v1alpha1/httpchaos_types.go:24]: Please remove this comment `+kubebuilder:printcolumn:name="action"` with this situation :)

**Review by STRRL — APPROVED** [review]
LGTM! Thanks for your contribution!

**Comment by STRRL** [comment]
PTAL

/cc @YangKeao @iguoyr @WangXiangUSTC @cwen0

**Comment by cwen0** [comment]
If the duration field is empty, what happens?

**Comment by STRRL** [comment]
> If the duration field is empty, what happens?

`kubectl` would show an empty string as the value for this column.

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> cherry pick to release-2.0 failed

I would cherrypick this PR into release-2.0 manually.

**Pull request merged**